packages/commerce-sdk-react/src/provider.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React, {ReactElement} from 'react'
+
+import {
+    ShopperBaskets,
+    ShopperContexts,
+    ShopperCustomers,
+    ShopperLogin,
+    ShopperOrders,
+    ShopperProducts,
+    ShopperPromotions,
+    ShopperDiscoverySearch,
+    ShopperGiftCertificates,
+    ShopperSearch
+} from 'commerce-sdk-isomorphic'
+
+import {ApiClients} from './hooks/types'
+
+export interface CommerceApiProviderProps {
+    children: React.ReactNode
+    clientId: string
+    organizationId: string
+    siteId: string
+    shortCode: string
+    proxy: string
+    locale: string
+    currency: string
+}
+
+export const CommerceApiContext = React.createContext({} as ApiClients)
```

Comment:
This is the case of "build objects up as we go". `React.createContext` requires an initial value to be passed in. In our case, it will be the set of api clients. But we can't initialize them until later.

> Can we guarantee that the context will be replaced with an actual ApiClients object before anyone uses it?

Yes, the context will have the actual ApiClients object:
- The Context Provider will be mounted first and will initialize the api clients
- Then the children component will be mounted. So if it calls `useCommerceApi` to consume the context, the api clients will already be initialized.